File: 7.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	pos := readInts(scanner.Text())
	maxpos := maxSlice(pos)
	fuel := make([]int64, 0, maxpos+1)
	fuel2 := make([]int64, 0, maxpos+1)
	for p := int64(0); p <= maxpos; p++ {
		sum, sum2 := int64(0), int64(0)
		for _, i := range pos {
			sum += abs(i - p)
			sum2 += (abs(i-p) + 1) * (i + p - 2*min(i, p)) / 2
		}
		fuel = append(fuel, sum)
		fuel2 = append(fuel2, sum2)
	}
	fmt.Println(minSlice(fuel), minSlice(fuel2))
}

File: 7.rb
#!ruby
pos = $stdin.gets.strip.scan(/\d+/).map(&:to_i)
fuel, fuel2 = Array.new(pos.max, 0), Array.new(pos.max, 0)
for p in 0..pos.max
    fuel[p] = pos.map do |pp| (pp - p).abs end.sum
    fuel2[p] = pos.map do |pp| ((pp - p).abs + 1) * (pp + p - 2 * [pp, p].min) / 2 end.sum
end
puts(fuel.min, fuel2.min)